src/server/network/network.ts

```diff
@@ -2,64 +2,72 @@ import dgram = require('dgram');
+export class UDPServers {
+  private servers: Array<UDPServer>;
   private clients: Map<string, ClientListener>;
 
   constructor(teams: Array<TeamData>) {
     this.clients = new Map<string, ClientListener>();
-    this.listeners = new Array<Listener>();
+    this.servers = new Array<UDPServer>();
 
     teams.forEach((team: TeamData) => {
-      // Determine if we are listening on a IPv4 or IPv6 address and resolve any DNS names
+      // Determine if we are listening on a IPv4 or IPv6 address and resolve any DNS address
+      // If the specified address can be resolved to an IPv4 address prefer that over an IPv6 address
       dns.lookup(
         team.address,
-        { family: 0, all: false },
+        { family: 0, all: true },
         (
           error: NodeJS.ErrnoException | null,
-          address: string,
-          family: number,
+          addresses: dns.LookupAddress[],
         ) => {
-          if (family === 6) {
-            const listener = {
-              server: dgram.createSocket({ type: 'udp6', reuseAddr: true }),
-              ipv6: true,
-              team: {
-                name: team.name,
-                address: address,
-                port: team.port,
-              },
-            };
-            this.initialiseListener(listener);
-            this.listeners.push(listener);
-          } else if (family === 4) {
-            const listener = {
-              server: dgram.createSocket({ type: 'udp4', reuseAddr: true }),
-              ipv6: false,
-              team: {
-                name: team.name,
-                address: address,
-                port: team.port,
-              },
-            };
-            this.initialiseListener(listener);
-            this.listeners.push(listener);
+          // Find the first IPv4 and IPv6 addresses that were returned
+          let ipv4 = '';
+          let ipv6 = '';
+          addresses.forEach((address: dns.LookupAddress) => {
+            if (address.family === 6 && ipv6 === '') {
+              ipv6 = address.address;
+            }
+            if (address.family === 4 && ipv4 === '') {
+              ipv4 = address.address;
+            }
+          });
+
+          // Prefer IPv4 addresses
+          if (ipv4 !== '') {
+            this.servers.push(
+              UDPServer.of(
+                { name: team.name, address: ipv4, port: team.port },
+                (team: TeamData, payload: Buffer, rinfo: dgram.RemoteInfo) => {
+                  this.onMessage(team, payload, rinfo);
+                },
+                false,
+              ),
+            );
+          } else if (ipv6 !== '') {
+            this.servers.push(
+              UDPServer.of(
+                { name: team.name, address: ipv6, port: team.port },
+                (team: TeamData, payload: Buffer, rinfo: dgram.RemoteInfo) => {
+                  this.onMessage(team, payload, rinfo);
+                },
+                true,
+              ),
+            );
```

Comment:
```suggestion
          if (ipv4.length > 0 || ipv6.length > 0) {
            this.servers.push(
              UDPServer.of(
                { name: team.name, address: ipv4.length > 0 ? ipv4[0] : ipv6[0], port: team.port },
                (team: TeamData, payload: Buffer, rinfo: dgram.RemoteInfo) => {
                  this.onMessage(team, payload, rinfo);
                },
                false,
              ),
            );
```
And don't bother storing ipv4 vs ipv6, just check what the address is later to do the multicast checking